DAOS-16754 build: Add support for http proxy

htjain opened the pull request:
Added support for http proxy in DAOS docker build infra.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

htjain committed:
DAOS-16754 build: Add support for http proxy

Added support for http proxy in DAOS docker build infra.

Signed-off-by: Hemant Jain <46592577+htjain@users.noreply.github.com>

brianjmurrell commented:
@htjain Can you please resubmit your commit with a `Signed-off-by:`?  That will be required in order to contribute your change to the project.

brianjmurrell commented:
@htjain I should have also requested that your commit(s) include the pragma: `Run-GHA: true`.  This enables running the full suite of testing.  I'll push a commit to this PR to effect that but please add that to any future commits you push to this or any other PR.  Thanks!

brianjmurrell committed:
Run CI in GitHub Actions

Run-GHA: true

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell committed:
Remove erroneously added file

Not adding Run-GHA: since this is a NOOP commit just to remove a
previously added inconsequential file.  No need to waste CI time on it.

Required-githooks: true

Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

brianjmurrell commented:
@htjain If you think this is ready for review and landing, you should remove the _draft_ status.  People typically won't review drafts unless specifically asked (by more than just adding as a reviewer).  Click this button to move it from draft to being ready for review: 
![image](https://github.com/user-attachments/assets/5ff16136-6324-4944-8faa-d6bd513a696c)
